packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -0,0 +1,783 @@
+    final FocusNode focusedChild = nearestScope.focusedChild;
+    if (focusedChild == null) {
+      findFirstFocus(currentNode).requestFocusFromPolicy();
+      return true;
+    }
+    final List<FocusNode> sortedNodes = _sortByGeometry(nearestScope).toList();
+    if (forward && focusedChild == sortedNodes.last) {
+      sortedNodes.first.requestFocusFromPolicy();
+      return true;
+    }
+    if (!forward && focusedChild == sortedNodes.first) {
+      sortedNodes.last.requestFocusFromPolicy();
+      return true;
+    }
+
+    final Iterable<FocusNode> maybeFlipped = forward ? sortedNodes : sortedNodes.reversed;
+    FocusNode previousNode;
+    for (FocusNode node in maybeFlipped) {
+      if (previousNode == focusedChild) {
+        node.requestFocusFromPolicy();
+        return true;
+      }
+      previousNode = node;
+    }
+    return false;
+  }
+
+  @override
+  bool next(FocusNode currentNode) => _move(currentNode, forward: true);
+
+  @override
+  bool previous(FocusNode currentNode) => _move(currentNode, forward: false);
+}
+
+/// A widget that describes an inherited focus policy for focus traversal.
+///
+/// By default, traverses in widget order using
+/// [WidgetOrderFocusTraversalPolicy].
+///
+/// See also:
+///
+///   * [FocusNode], for a description of the focus system.
+///   * [WidgetOrderFocusTraversalPolicy], a policy that relies on the widget
+///     creation order to describe the order of traversal.
+///   * [ReadingOrderTraversalPolicy], a policy that describes the order as the
+///     natural "reading order" for the current [Directionality].
+///   * [DirectionalFocusTraversalPolicyMixin] a mixin class that implements
+///     focus traversal in a direction.
+class DefaultFocusTraversal extends InheritedWidget {
+  /// Creates a FocusTraversal object.
+  ///
+  /// The [policy] and [child] arguments must not be null.
+  const DefaultFocusTraversal({
+    Key key,
+    this.policy = const WidgetOrderFocusTraversalPolicy(),
+    @required Widget child,
+  })  : assert(policy != null),
+        super(key: key, child: child);
+
+  /// The policy used to move the focus from one focus node to another.
+  ///
+  /// By default, traverses in widget order using
+  /// [WidgetOrderFocusTraversalPolicy].
+  ///
+  /// See also:
+  ///
+  ///  * [FocusTraversalPolicy] for the API used to impose traversal order
+  ///    policy.
+  ///  * [WidgetOrderFocusTraversalPolicy] for a traversal policy that traverses
+  ///    nodes in the order they are added to the widget tree.
+  ///  * [ReadingOrderTraversalPolicy] for a traversal policy that traverses
+  ///    nodes in the reading order defined in the widget tree, and then top to
+  ///    bottom.
+  final FocusTraversalPolicy policy;
+
+  /// Returns the [DefaultFocusTraversal] that most tightly encloses the given
+  /// [BuildContext].
+  ///
+  /// The [context] argument must not be null.
+  static FocusTraversalPolicy of(BuildContext context, {bool ifAny = false}) {
```

Comment:
There are a few of these already and they're `{ bool nullOk = false }`

Reply:
Fixed.